client/consensus/src/collators/basic.rs

```diff
@@ -0,0 +1,248 @@
+    SO: SyncOracle + Send + Sync + Clone + 'static,
+    Proposer: ProposerInterface<Block> + Send + Sync + 'static,
+    CS: CollatorServiceInterface<Block> + Send + Sync + 'static,
+    P: Pair,
+    P::Public: AppPublic + Member + Codec,
+    P::Signature: TryFrom<Vec<u8>> + Member + Codec,
+    GOH: RetrieveAuthoritiesFromOrchestrator<
+            Block,
+            (PHash, PersistedValidationData),
+            Vec<AuthorityId<P>>,
+        >
+        + 'static
+        + Sync
+        + Send,
+{
+    async move {
+        let mut collation_requests = match params.collation_request_receiver {
+            Some(receiver) => receiver,
+            None => {
+                cumulus_client_collator::relay_chain_driven::init(
+                    params.collator_key,
+                    params.para_id,
+                    params.overseer_handle,
+                )
+                .await
+            }
+        };
+
+        let mut collator = {
+            let params = collator_util::Params {
+                create_inherent_data_providers: params.create_inherent_data_providers.clone(),
+                block_import: params.block_import,
+                relay_client: params.relay_client.clone(),
+                keystore: params.keystore.clone(),
+                para_id: params.para_id,
+                proposer: params.proposer,
+                collator_service: params.collator_service,
+            };
+
+            collator_util::Collator::<Block, P, _, _, _, _, _>::new(params)
+        };
+
+        while let Some(request) = collation_requests.next().await {
+            macro_rules! reject_with_error {
+				($err:expr) => {{
+					request.complete(None);
+					tracing::error!(target: crate::LOG_TARGET, err = ?{ $err });
+					continue;
+				}};
+			}
+
+            macro_rules! try_request {
+                ($x:expr) => {{
+                    match $x {
+                        Ok(x) => x,
+                        Err(e) => reject_with_error!(e),
+                    }
+                }};
+            }
+
+            let validation_data = request.persisted_validation_data();
+
+            let parent_header = try_request!(Block::Header::decode(
+                &mut &validation_data.parent_head.0[..]
+            ));
+
+            let parent_hash = parent_header.hash();
+
+            if !collator
+                .collator_service()
+                .check_block_status(parent_hash, &parent_header)
+            {
+                continue;
+            }
+
+            let relay_parent_header = match params
+                .relay_client
+                .header(RBlockId::hash(*request.relay_parent()))
+                .await
+            {
+                Err(e) => reject_with_error!(e),
+                Ok(None) => continue, // sanity: would be inconsistent to get `None` here
+                Ok(Some(h)) => h,
+            };
+
+            let authorities = match params
```

Comment:
Add a comment here of the sort: retrieve the authorities that are able to produce block